apps/dh/api-dh/source/DataHub.WebApi/GraphQL/Types/Calculation/CalculationType.cs

```diff
@@ -55,7 +55,57 @@ protected override void Configure(IObjectTypeDescriptor<CalculationDto> descript
                 CalculationState.Completed => ProcessStatus.Success,
                 CalculationState.Failed => ProcessStatus.Danger,
                 CalculationState.Executing => ProcessStatus.Info,
-                _ => ProcessStatus.Info,
+            });
+
+        descriptor
+            .Field("progress")
+            .Type<NonNullType<ListType<NonNullType<ObjectType<CalculationProgress>>>>>()
+            .Resolve(context =>
+            {
+                var state = context.Parent<CalculationDto>().OrchestrationState;
+                return new List<CalculationProgress>
+                {
+                    GetCalculationProgress(CalculationProgressStep.Schedule, state),
+                    GetCalculationProgress(CalculationProgressStep.Calculate, state),
+                    GetCalculationProgress(CalculationProgressStep.ActorMessageEnqueue, state),
+                };
             });
     }
+
+    private static CalculationProgress GetCalculationProgress(
+        CalculationProgressStep step,
+        CalculationOrchestrationState state) =>
+        step switch
+        {
+            CalculationProgressStep.Schedule => state switch {
+                CalculationOrchestrationState.Scheduled => new(step, CalculationProgressStatus.Pending, true),
+                CalculationOrchestrationState.Calculating => new(step, CalculationProgressStatus.Completed, false),
+                CalculationOrchestrationState.CalculationFailed => new(step, CalculationProgressStatus.Completed, false),
```

Comment:
Suggestion: Should this not be CalculationProgressStatus.Failed?

Reply:
No, but I understand why the logic is a bit tricky to follow. There are 3 steps, `Schedule`, `Calculate` and `ActorMessageEnqueue`. That particular line gets the status of the first step (`Schedule`) based on the value of `CalculationOrchestrationState`. If the value of `CalculationOrchestrationState` is `CalculationFailed`, then it is implied that the first step must be completed, since the calculation cannot start before it is scheduled.

Take the following screenshot of the progress tracker (which is what this field is going to be used for). This would be how it should look if the state is `CalculationOrchestrationState.CalculationFailed`. Notice how the first step (Schedule) is completed. It is only the second step (Calculate) that should have the failed status.

![progress-tracker](https://github.com/Energinet-DataHub/greenforce-frontend/assets/22732929/7fc57ceb-5bde-4ca7-9fb6-35fa9f47b6a5)

Reply:
But if you have any ideas on how to make this logic more clear, then that would be awesome!